### PR TITLE
Amanda

### PR DESCRIPTION
**Pull request by ajtsao1**
bothLigands
TAM_react

I'm not completely sure if this is the correct way to use view in place of a pointer.
I don't think my branch was set up to track the master so some of the new code that I didn't add may not show up.

### CODE REVIEW

File: src/bothLigands.jl [hunk 1]
```diff
@@ -20,3 +20,28 @@ mutable struct Lsrates{T}
...
+function TAM_react(const R::T, const dR::T)
+    tr = view(pparams,:,:)
+    
+    react_module(R, dR, tr.curL) # We give the same internal ligand address as below because it will be overwritten
+    react_module(R+iR, dR+iR, {{R[15+iR]/tr.internalV, R[16+iR]/tr.internalV}})
+
+    dR[1] += tr.expression
+
+    for UInt i = 1:8
+        trafFunc(@view dR[i], @view dR[i+iR], tr.internalize, R[i], R[i+iR], tr.kRec, tr.kDeg, tr.fElse, tr.internalFrac)
```
Comment (aarmey) [inline src/bothLigands.jl]: Rather than a loop you can use a slice—e.g. `dR[1:8]` instead of `dR[i]`.

File: src/bothLigands.jl [hunk 2]
```diff
@@ -20,3 +20,28 @@ mutable struct Lsrates{T}
...
+    end
+    
+    for UInt i = 9:13
+        trafFunc(@view dR[i], @view dR[i+iR], tr.pYinternalize, R[i], R[i+iR], tr.kRec, tr.kDeg, tr.fElse, tr.internalFrac)
```
Comment (aarmey) [inline src/bothLigands.jl]: Same here.

File: src/bothLigands.jl [hunk 3]
```diff
@@ -20,3 +20,28 @@ mutable struct Lsrates{T}
...
+# @param[in] R  Double vector for current state of model
+# @param dR  Double vector for returning derivative of current state
+# @return Always returns 0 for success
+
+
+function TAM_react(const R::T, const dR::T)
+    tr = view(pparams,:,:)
+    
+    react_module(R, dR, tr.curL) # We give the same internal ligand address as below because it will be overwritten
+    react_module(R+iR, dR+iR, {{R[15+iR]/tr.internalV, R[16+iR]/tr.internalV}})
```
Comment (aarmey) [inline src/bothLigands.jl]: Believe a tuple is `(A, B)`, not `{{A, B}}`

File: src/bothLigands.jl [hunk 4]
```diff
@@ -20,3 +20,28 @@ mutable struct Lsrates{T}
...
 end
+
+# @brief Callback function from CVode to handle ODE solving
+# @param[in] R  Double vector for current state of model
+# @param dR  Double vector for returning derivative of current state
+# @return Always returns 0 for success
+
+
+function TAM_react(const R::T, const dR::T)
+    tr = view(pparams,:,:)
```
Comment (aarmey) [inline src/bothLigands.jl]: What are you trying here?

File: src/TAMode.jl [hunk 5]
```diff
@@ -54,7 +54,7 @@ function runTAM(tps::Array{Float64,1}, params, gasStim::Float64)::Array{Float64,
     end
 
     return runTAMinit(tps, params, solInit)
-end
+    end # HI
```
Comment (aarmey) [inline src/TAMode.jl]: Remove.

**Comment by ajtsao1** [comment]
@aarmey I wasn't sure if I should open a new pull request but I pushed the changes you requested for TAM_react. I'm still familiarizing myself with Github but I believe you should be able to see the changes.

**Pull request merged**